central/graphql/resolvers/node_vulnerabilities_test.go

```diff
@@ -0,0 +1,600 @@
+	nodeCVEPostgres.Destroy(s.ctx, s.db)
+	nodeComponentCVEEdgePostgres.Destroy(s.ctx, s.db)
+
+	// create mock resolvers, set relevant ones
+	s.resolver = NewMock()
+
+	// nodeCVE datastore
+	nodeCVEStore := nodeCVEPostgres.CreateTableAndNewStore(s.ctx, s.db, s.gormDB)
+	nodeCVEIndexer := nodeCVEPostgres.NewIndexer(s.db)
+	nodeCVESearcher := nodeCVESearch.New(nodeCVEStore, nodeCVEIndexer)
+	nodeCVEDatastore, err := nodeCVEDataStore.New(nodeCVEStore, nodeCVEIndexer, nodeCVESearcher, concurrency.NewKeyFence())
+	s.NoError(err, "Failed to create nodeCVEDatastore")
+	s.resolver.NodeCVEDataStore = nodeCVEDatastore
+
+	// node datastore
+	riskMock := mockRisks.NewMockDataStore(gomock.NewController(s.T()))
+	nodeStore := nodePostgres.CreateTableAndNewStore(s.ctx, s.T(), s.db, s.gormDB, false)
+	nodeIndexer := nodePostgres.NewIndexer(s.db)
+	nodeSearcher := nodeSearch.NewV2(nodeStore, nodeIndexer)
+	nodePostgresDataStore := nodeDackboxDataStore.NewWithPostgres(nodeStore, nodeIndexer, nodeSearcher, riskMock, ranking.NewRanker(), ranking.NewRanker())
+	nodeGlobalDatastore, err := nodeGlobalDataStore.New(nodePostgresDataStore)
+	s.NoError(err, "Failed to create nodeGlobalDatastore")
+	s.resolver.NodeGlobalDataStore = nodeGlobalDatastore
+
+	// nodeComponent datastore
+	nodeCompStore := nodeComponentPostgres.CreateTableAndNewStore(s.ctx, s.db, s.gormDB)
+	nodeCompIndexer := nodeComponentPostgres.NewIndexer(s.db)
+	nodeCompSearcher := nodeComponentSearch.New(nodeCompStore, nodeCompIndexer)
+	s.resolver.NodeComponentDataStore = nodeComponentDataStore.New(nodeCompStore, nodeCompIndexer, nodeCompSearcher, riskMock, ranking.NewRanker())
+
+	// nodeComponentCVEEdge datastore
+	nodeComponentCveEdgeStore := nodeComponentCVEEdgePostgres.CreateTableAndNewStore(s.ctx, s.db, s.gormDB)
+	nodeCompontCveEdgeIndexer := nodeComponentCVEEdgePostgres.NewIndexer(s.db)
+	nodeComponentCveEdgeSearcher := nodeComponentCVEEdgeSearch.New(nodeComponentCveEdgeStore, nodeCompontCveEdgeIndexer)
+	nodeComponentCveEdgeDatastore, err := nodeComponentCVEEdgeDataStore.New(nodeComponentCveEdgeStore, nodeCompontCveEdgeIndexer, nodeComponentCveEdgeSearcher)
+	s.NoError(err)
+	s.resolver.NodeComponentCVEEdgeDataStore = nodeComponentCveEdgeDatastore
+
+	// Sac permissions
+	s.ctx = sac.WithAllAccess(s.ctx)
+
+	// loaders used by graphql layer
+	loaders.RegisterTypeFactory(reflect.TypeOf(storage.Node{}), func() interface{} {
+		return loaders.NewNodeLoader(nodePostgresDataStore)
+	})
+	loaders.RegisterTypeFactory(reflect.TypeOf(storage.NodeComponent{}), func() interface{} {
+		return loaders.NewNodeComponentLoader(s.resolver.NodeComponentDataStore)
+	})
+	loaders.RegisterTypeFactory(reflect.TypeOf(storage.NodeCVE{}), func() interface{} {
+		return loaders.NewNodeCVELoader(s.resolver.NodeCVEDataStore)
+	})
+	s.ctx = loaders.WithLoaderContext(s.ctx)
+
+	// Add Test Data to DataStores
+	testNodes := testNodes()
+	for _, node := range testNodes {
+		err = nodeStore.Upsert(s.ctx, node)
```

Comment:
store must not be accessed directly. Most of our business logic lies in datastores. Call upsert on datastore.

Reply:
Similarly for image test.